lauraguerra1/whats-cookin
# Feature/add settings panel

## Description
Adds new settings panel with chart data of the top 10 most clicked recipes!
<img width="954" alt="Screenshot 2023-06-03 at 10 46 12 PM" src="https://github.com/lauraguerra1/whats-cookin/assets/32660020/609cf824-e057-450e-88c5-406410fe1957">

I used [chart.js](https://www.chartjs.org/) to visualize the data.


## Testing Notes
- run `npm install` before testing so that the chart.js package is brought in
- clicking on the settings gear now opens a new admin modal displaying the top 10 most clicked recipes
  - ensure they are listed in correct descending order
  - Click on a few recipes to increase their hit totals, then relook at the hits chart to ensure the data is getting properly updated and visualized
  - I refactored some of the view recipe code, so make sure all view recipe flows still function as expected
- Please let me know about any concerns/ideas/requests for the chart display as there are a lot of ways we could alter or enhance.
- i DID use some regex to split long recipe names into smaller chunks so their labels would word-wrap on the chart.  If you'd prefer I can definitely do it with iterators.
